information_for_beamlines_ds/information_for_beamlines.py

```diff
@@ -0,0 +1,173 @@
+        """
+        Docstring for 'safe_init_device' - it is just safe initialization of the DS.
+        :return:
+        """
+        super(InformationForBeamlines, self).safe_init_device()
+        self.set_state(DevState.ON)
+        self.set_status("Device is running")
+        self.general_info = 'No information.'
+        self.filling_pattern = 'No filling pattern.'
+
+    # ----------
+    # Attributes
+    # ----------
+
+    GeneralInfo = attribute(
+        dtype='str',
+        access=AttrWriteType.READ_WRITE,
+        label="General info",
+        doc="General message from CR to all beamlines.",
+    )
+
+    FillingPattern = attribute(
+        dtype='str',
+        access=AttrWriteType.READ_WRITE,
+        label="Filling Pattern",
+        doc="The filling pattern currently used in the storage ring.",
+    )
+
+    # ----------
+    # Proxy attributes
+    # ----------
+
+    InjectionStatus = proxy_attribute(
+        dtype='bool',
+        property_name="InjectionStatusTag",
+        display_level=DispLevel.EXPERT,
+        access=AttrWriteType.READ,
+        doc="Forwarded injection status tag from MPS PLC."
+    )
+
+    ExperimentEnable = proxy_attribute(
+        dtype='bool',
+        property_name='ExperimentEnableTag',
+        display_level=DispLevel.EXPERT,
+        access=AttrWriteType.READ,
+        doc='Forwarded experiment enable status tag from MPS PLC.'
+    )
+
+    BeamCurrent = proxy_attribute(
+        dtype='float',
+        property_name='BeamCurrentAttr',
+        unit='A',
+        format='%8.6f',
+        access=AttrWriteType.READ,
+        display_level=DispLevel.EXPERT,
+        doc="Forwarded beam current from BIM."
+    )
+
+    BIMState = proxy_attribute(
+        dtype="DevState",
+        property_name="BIMStateTag",
+        access=AttrWriteType.READ,
+        display_level=DispLevel.EXPERT,
+        doc="State of the beam intensity monitor."
+    )
+
+    MPSState = proxy_attribute(
+        dtype="DevState",
+        property_name="MPSStateTag",
+        access=AttrWriteType.READ,
+        display_level=DispLevel.EXPERT,
+        doc="State of the MPS PLC device."
+    )
+
+    # ----------
+    # Logical attributes
+    # ----------
+
+    @logical_attribute(
+        dtype='str',
+        label="Operation Status",
+        bind=["BeamCurrent","ExperimentEnable","InjectionStatus"],
+        doc="One of: Injection, MDT (Machine Dedicated Time), Experiment Enable.",)
+    def OperationStatus(self, beam_c,exp_en,inj_stat):
+        if inj_stat:
+            return 'Injection'
+        elif beam_c > 1 and exp_en:
+            return 'Experiment Enable'
+        else:
+            return 'MDT'
+
+    @logical_attribute(
+        dtype=float,
+        unit='mA',
+        bind=["BeamCurrent"],
+        format='%4.3f',
+        label='Beam Current (mA)',
+        doc='Beam current in miliamperes.',
+        display_level=DispLevel.EXPERT)
+    def BeamCurrent_mA(self, data):
```

Comment:
Please remove the `BeamCurrent_mA` attribute and include the calculation in the `BeamCurrent` attribute.
We want to have only one attribute, called `BeamCurrent` which displays beam current in milliamperes.